iothub_client/samples/iothub_client_sample_x509/iothub_client_sample_x509.c

```diff
@@ -19,15 +19,15 @@ and removing calls to _DoWork will yield the same results. */
 #include "azure_c_shared_utility/crt_abstractions.h"
 #include "azure_c_shared_utility/platform.h"
 
-#ifdef USE_MQTT
-#include "iothubtransportmqtt.h"
-#endif
 #ifdef USE_HTTP
 #include "iothubtransporthttp.h"
 #endif
 #ifdef USE_AMQP
 #include "iothubtransportamqp.h"
 #endif
+#ifdef USE_MQTT
```

Comment:
USE_HTTP is set in CMAKELIST and you need to change it to USE_MQTT if you want protocols to default to MQTT.